mirgecom/simutil.py

```diff
@@ -186,7 +186,8 @@ def check_range_local(discr, dd, field, min_value, max_value):
 
 def check_naninf_local(discr, dd, field):
     """Check for any NANs or Infs in the field."""
-    s = op.nodal_sum_loc(discr, dd, field)
+    actx = field.array_context
+    s = actx.to_numpy(op.nodal_sum_loc(discr, dd, field))
```

Comment:
Will this be expensive to do on the host?

Reply:
Nope. On the host we are simply checking if the scalar is `inf` or `nan`. The sum-reduction which operates on the vector would still happen on the device.